libs/d3d12/main.c

```diff
@@ -0,0 +1,377 @@
+        return FALSE;
+
+    if (!(properties = calloc(count, sizeof(*properties))))
+        return FALSE;
+
+    if (pfn_vkEnumerateInstanceExtensionProperties(NULL, &count, properties) >= 0)
+    {
+        for (i = 0; i < count; ++i)
+        {
+            if (!strcmp(properties[i].extensionName, name))
+            {
+                ret = TRUE;
+                break;
+            }
+        }
+    }
+
+    free(properties);
+    return ret;
+}
+
+static VkPhysicalDevice d3d12_get_vk_physical_device(struct vkd3d_instance *instance,
+        PFN_vkGetInstanceProcAddr pfn_vkGetInstanceProcAddr, struct DXGI_ADAPTER_DESC *adapter_desc)
+{
+    PFN_vkGetPhysicalDeviceProperties2 pfn_vkGetPhysicalDeviceProperties2 = NULL;
+    PFN_vkEnumeratePhysicalDevices pfn_vkEnumeratePhysicalDevices;
+    VkPhysicalDevice vk_physical_device = VK_NULL_HANDLE;
+    VkPhysicalDeviceIDProperties id_properties;
+    VkPhysicalDeviceProperties2 properties2;
+    VkPhysicalDevice *vk_physical_devices;
+    VkInstance vk_instance;
+    unsigned int i;
+    uint32_t count;
+    VkResult vr;
+
+    vk_instance = vkd3d_instance_get_vk_instance(instance);
+
+    pfn_vkEnumeratePhysicalDevices = (void *)pfn_vkGetInstanceProcAddr(vk_instance, "vkEnumeratePhysicalDevices");
+
+    if (check_vk_instance_extension(vk_instance, pfn_vkGetInstanceProcAddr, VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME))
+        pfn_vkGetPhysicalDeviceProperties2 = (void *)pfn_vkGetInstanceProcAddr(vk_instance, "vkGetPhysicalDeviceProperties2KHR");
+
+    if ((vr = pfn_vkEnumeratePhysicalDevices(vk_instance, &count, NULL)) < 0)
+    {
+        WARN("Failed to get device count, vr %d.\n", vr);
+        return VK_NULL_HANDLE;
+    }
+    if (!count)
+    {
+        WARN("No physical device available.\n");
+        return VK_NULL_HANDLE;
+    }
+
+    if (!(vk_physical_devices = calloc(count, sizeof(*vk_physical_devices))))
+        return VK_NULL_HANDLE;
+
+    if ((vr = pfn_vkEnumeratePhysicalDevices(vk_instance, &count, vk_physical_devices)) < 0)
+        goto done;
+
+    if (pfn_vkGetPhysicalDeviceProperties2)
+    {
+        TRACE("Matching adapters by LUIDs.\n");
+
+        for (i = 0; i < count; ++i)
+        {
+            memset(&id_properties, 0, sizeof(id_properties));
+            id_properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ID_PROPERTIES;
+
+            properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
+            properties2.pNext = &id_properties;
+
+            pfn_vkGetPhysicalDeviceProperties2(vk_physical_devices[i], &properties2);
+
+            if (!memcmp(id_properties.deviceLUID, &adapter_desc->AdapterLuid, VK_LUID_SIZE))
+            {
+                vk_physical_device = vk_physical_devices[i];
+                break;
+            }
+        }
+    }
+
+    if (!vk_physical_device)
+    {
+        FIXME("Could not find Vulkan physical device for DXGI adapter.\n");
+        WARN("Using first available physical device...\n");
```

Comment:
Maybe for some big brain fallback we could try to find a device with matching vendor/device ID, like DXVK does.

Reply:
Resolved locally.